plugins/ptp_operator/metrics/registry.go

```diff
@@ -171,3 +191,13 @@ func DeleteInterfaceRoleMetrics(process, ptpInterface string) {
 	InterfaceRole.Delete(prometheus.Labels{
 		"process": process, "node": ptpNodeName, "iface": ptpInterface})
 }
+
+// UpdateProcessStatusMetrics  -- update process status metrics
+func UpdateProcessStatusMetrics(process, cfgName string, status int64) {
+	ProcessStatus.With(prometheus.Labels{
+		"process": process, "node": ptpNodeName, "config": cfgName}).Set(float64(status))
+	if status == PtpProcessUp {
```

Comment:
Since there is no caching of previous status, if the process stays up and we keep checking the logs, this `restart count` will keep increasing even though there is no restart. The logic does not check out.

Reply:
The logs are read in real-time. We don't go back in time, just read the present time.